cpp/src/column/column_view.cpp

```diff
@@ -144,9 +144,9 @@ column_view logical_cast(column_view const& input, data_type type)
                      input._children};
```

Comment:
Are we are including compound types in bit-castable?
Technically, a dictionary could be bit-castable if they key-type is. For example an FLOAT32 key could be cast to INT32 but just changing the dictionary's child keys column_view. This could work for lists columns too I suppose.

If this is out of scope, I think adding a check for `cudf::is_compound` may be needed in `is_bit_castable`. This could take the place of the specific test for `string_view` too.

Reply:
Since this is called `bit_cast`, I tried to follow the rules of `std::bit_cast`, so the converting types should be trivially copyable:  
```
template <class To, class From>
typename std::enable_if_t<
    sizeof(To) == sizeof(From) &&
    std::is_trivially_copyable_v<From> &&
    std::is_trivially_copyable_v<To>,
    To>
// constexpr support needs compiler magic
bit_cast(const From& src) noexcept
{
    static_assert(std::is_trivially_constructible_v<To>,
        "This implementation additionally requires destination type to be trivially constructible");
 
    To dst;
    std::memcpy(&dst, &src, sizeof(To));
    return dst;
}
```